### PR TITLE
Move EUnit to CT

### LINKED ISSUE #101: Move tests to a CT suite
See discussion on https://github.com/elli-lib/elli/pull/90

### PR DESCRIPTION
**Pull request by paulo-ferraz-oliveira**
Closes #101.

Depends on the CI file updates from #114, so shall be rebased and updated (if required) after that one.

In CT (this pull request):

```
  |----------------------------------|------------|
  |                          module  |  coverage  |
  |----------------------------------|------------|
  |                            elli  |       77%  |
  |           elli_example_callback  |       90%  |
  |  elli_example_callback_handover  |      100%  |
  |         elli_example_middleware  |      100%  |
  |                       elli_http  |       67%  |
  |                 elli_middleware  |      100%  |
  |        elli_middleware_compress  |      100%  |
  |                    elli_request  |       78%  |
  |                   elli_sendfile  |       62%  |
  |                        elli_tcp  |       86%  |
  |                       elli_test  |      100%  |
  |                       elli_util  |      100%  |
  |----------------------------------|------------|
  |                           total  |       76%  |
  |----------------------------------|------------|
```

In EUnit (`main`)

```
  |----------------------------------|------------|
  |                          module  |  coverage  |
  |----------------------------------|------------|
  |                            elli  |       77%  |
  |           elli_example_callback  |       90%  |
  |  elli_example_callback_handover  |      100%  |
  |         elli_example_middleware  |      100%  |
  |                       elli_http  |       68%  |
  |                 elli_middleware  |      100%  |
  |        elli_middleware_compress  |      100%  |
  |                    elli_request  |       78%  |
  |                   elli_sendfile  |       62%  |
  |                        elli_tcp  |       86%  |
  |                       elli_test  |      100%  |
  |                       elli_util  |      100%  |
  |----------------------------------|------------|
  |                           total  |       76%  |
  |----------------------------------|------------|
```

I tracked the differences for the `elli_http` module and they're only related to the missing test code (7 lines in the previous code). In `elli_test` this is not noticeable because we covered 100% and now cover 100% still...

From the commit notes:

```
Move tests from EUnit to Common Test
Notice the use of README.md from the linked `priv` folder

Files were mostly renamed and then:
- added init/end_per_suite (from setup/teardown)
- added init/end_per_testcase (from elli_test_)
- removed _test suffix from exported function names
- moved rebar.config test options to test profile
```

```
Add a shareable (example + tests) symbolic link
Used as filename:join(code:priv_dir(elli), "README.md") for
single reference...
```

```
Prepare for CT
- deps and ebin are rebar 2
- ignore *.beam is not required, since _build
- .eunit is legacy
- pre-hook lint (for eunit) is removed and made explicit in the CI file
- main.yml is tweaked to run static analysis first and tests later
```

### CODE REVIEW

**Comment by paulo-ferraz-oliveira** [comment]
~It's possible the tests fail for this one, and it also doesn't move up the supported OTP version. I'll copy that over from #112 and check what needs to be adapted.~

**Comment by paulo-ferraz-oliveira** [comment]
~CI failing for OTP 26. I'll try to replicate locally.~

**Comment by paulo-ferraz-oliveira** [comment]
~Removed OTP 26 from scope, since there're SSL -related issues to solve. Should be moved to a new PR, I guess...~

**Review by S-code652 — APPROVED** [review]

**Comment by paulo-ferraz-oliveira** [comment]
Is anything lacking for this one to get merged? I can then rebase and update the other ones.

**Comment by tsloughter** [comment]
Only issue with these from my perspective has been their size (even if it looks like many of the commits are the same across the PRs, its still that getting one merged means being confident with changes to a lot of files) and I just haven't had the time to look them over close enough to feel comfortable.

**Comment by paulo-ferraz-oliveira** [comment]
I understand. I'll try to rebase/squash, etc. to ease review and maybe that can remove some commits that moved back and forth... @tsloughter, I think I did what I wanted already, in all the pull requests I have opened. The first one needing merge will be #114, so the other ones can "inherit" the new maintenance range and updates to CI files.

**Comment by paulo-ferraz-oliveira** [comment]
Moved to draft. These don't need to be reviewed before #114, after which I'll rebase.

File: .github/workflows/main.yml
Comment (paulo-ferraz-oliveira) [inline .github/workflows/main.yml:1]: We go from `eunit+dialyzer` to `xref+dialyzer+lint+ct+cover`

**Note**: `eunit` was doing `lint`, as per `rebar.config`, but it's probably better to have it explicitly as a CI step.

**Comment by paulo-ferraz-oliveira** [comment]
@tsloughter

**Review by tsloughter — APPROVED** [review]

**Pull request merged**